doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,281 @@
+
+#### 3.3.3. ChassisStateDB Schema for DPU_STATE
+```
+Table: “DPU_STATE”
+
+SCHEMA
+key:  dpu_state:1
+ 
+HMSET dpu_state:1
+        "id": "1",    		#Key itself can be used?
+        "powered": "ON",
+        "pcie_link_state": "UP",
+        "pcie_link_time": "timestamp",
+        "pcie_link_reason": "up_down_related string",
+        "host_eth_link_state": "UP",
+        "host_eth_link_time": " timestamp ",
+        "host_eth_link_reason": "up_down_related string",
+        "firmware_state": "UP",
+        "firmware_time": " timestamp ",
+        "firmware_reason": ”gold boot a, ONIE version x",
+        "os_state": "UP",
+        "os_state_time": "timestamp",
+        "os_reason": ”version x",
+        "previos_reboot_reason_from_dpu": “Software reboot ”,
+        "previos_reboot_time_from_dpu ": “timestamp”,
+        “previous_reboot_reason_from_host”: ”Powered Down DPU - Temperature failure”,
+        "previos_reboot_time_from_host ": “timestamp”,
+        "control_plane_state": ”DOWN",
+        "control_plane_time": ”timestamp",
+        "control_plane_reason": ”containers restarting",
+        "data_plane_state": ”DOWN",
+        "data_plane_time": ”timestamp",
+        "data_plane_reason": ”Pipeline failure",
+```
+
+Besides the state and previous_reboot_reason_from_host other fields will be updated by the DPU once it boots.  The other fields will be updated by the switch from the information read from the hardware registers.
+
+### 3.4.   Midplane Interface
+
+A typical modular chassis includes a midplane-interface to interconnect the Supervisor & line-cards. The same design has been extended in case of a SmartSwitch. The mnic ethernet interface over PCIe which is the midplane-interface, interconnect the Switch Host and the DPUs.
+
+* When DPU card (SLED) or the Supervisor boots and as part of its initialization, midplane interface gets initialized.
+* Slot number is used in assigning an IP address to these interfaces.
+
+### 3.4.1.    DPU IP address allocation
+
+The table shows how the DPU-ID is used in assigning IP address to the midplane-interface endpoints for both the host and dpu end.
+
+
+| DPU_ID | DPU_Endpoint_IP | Host_Endpoint_IP | DPU_Endpoint_MAC | HOST_Endpoint_MAC |
+| --- | ---- | ------ | ------------------ | ------------------ |
+| 1 | 169.254.dpu_id.1 | 169.254.dpu_id.2 | DPU_MP_MAC1 | HOST_MP_MAC1 |
+| 2 | 169.254.2.1 | 169.254.2.2 | DPU_MP_MAC2 | HOST_MP_MAC2 |
+| 3 | 169.254.3.1 | 169.254.3.2 | DPU_MP_MAC3 | HOST_MP_MAC3 |
+| 4 | 169.254.4.1 | 169.254.4.2 | DPU_MP_MAC4 | HOST_MP_MAC4 |
+| 5 | 169.254.5.1 | 169.254.5.2 | DPU_MP_MAC5 | HOST_MP_MAC5 |
+| 6 | 169.254.6.1 | 169.254.6.2 | DPU_MP_MAC6 | HOST_MP_MAC6 |
+| 7 | 169.254.7.1 | 169.254.7.2 | DPU_MP_MAC7 | HOST_MP_MAC7 |
+| 8 | 169.254.8.1 | 169.254.8.2 | DPU_MP_MAC8 | HOST_MP_MAC8 |
+
+* The IP address allocation for the DPUs and the host can be static as in Modular chassis design 
```

Comment:
no DHCP?

Reply:
Looks like you are looking at an older version of the PR.  Please refer to the recent version